### DISCUSSION TITLE
Oled theme

### DISCUSSION THREAD

**Discussion by Toiletmason** [breezy-weather/breezy-weather#971] [Ideas]
### Describe your suggested feature

The black theme would be black.

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a short but informative title.
- [X] I have updated the app to **[latest version](https://github.com/breezy-weather/breezy-weather/releases/latest)**.

**Comment by Reed-Schimmel** [comment]
I would also love this feature

**Comment by Aturex1** [comment]
Me too, but I'm happy enough that there is a dark theme already

**Comment by codenyte** [comment]
This would be great

**Comment by Username1-a** [comment]
Would be greatly appreciated especially since in latest version (v5.0.3) you can't have Black OLED theme in Breezy Weather using Repainter app (probably because material components were updated)

**Comment by papjul** [comment]
Why isn't the current dark theme enough?

**Comment by Aturex1** [comment]
It is (at least for me) but it would be a nice extra to have a OLED-Theme for OLED-Smartphones

**Comment by codenyte** [comment]
OLED screens can achieve true blacks, which means that the background can be completely black without emitting any light. This contrast makes colors pop more and text appear clearer, reducing eye strain and improving the overall viewing experience. Theoretically, this could also save battery life, since a true black background does not require any backlight to be turned on. I personally really like Pitch black themes in mobile apps. There are many other examples of (FOSS) apps implementing a Pitch black/OLED theme. Maybe take a look at the Aurora Store, LibreTube, Xtra, AntennaPod and many more.

**Comment by Korb** [comment]
The requested feature is sometimes called "pure black background".

**Comment by papjul** [comment]
I'm opening this to contributions. Should be implemented as a new option of Dark mode preference:
- Follow system
- Light
- Dark
- Pure black

**Comment by codenyte** [comment]
Wouldn't it be better to just have a toggle for Pure Black below the theme selection menu? That way, the 'Follow system' setting could be used together with Pure black mode, i.e., when system-wide light mode is selected, Breezy Weather just uses light mode, and when dark mode is selected, it uses Pure black mode.

**Comment by papjul** [comment]
"Follow system" follows preference of the system which only offers light and dark to my knowledge. If there is a system preference for Pure black in standard Android that I missed, we can indeed automatically use it for the "Follow system" preference. Let me know if you find anything?

**Comment by codenyte** [comment]
You're right, there's no way to set system-wide pitch black mode. I just want to retain the ability to use 'Follow system' in combination with Pure black mode.

Some other apps do this, and it actually works like this:


|                   | Pure Black off | Pure Black on |
| ----------------- | -------------- | ------------- |
| System light mode | Light          | Light         |
| System dark mode  | Dark           | Pure black    |

**Comment by papjul** [comment]
Yes, I know, but I mentioned that I wanted to make it that way because I always found it was not a good thing:
- It creates the ability to choose a light theme with a "Pure black" toggle (or system preference which can be light, with "Pure black"), which doesn't make sense. We could play with disabled option for the light theme, but not with "follow system"
- It creates an isolated new option, which I would prefer avoiding; we already have a toggle for day/night mode
- If system preference is dark, but Breezy Weather applies pure black, technically, it doesn't follow system preference

I get what you're trying to achieve, but you all seemed to prefer OLED to reduce eye strain, so I find it curious that you now want both light and pure black at the same time?

Note that if you set Dark mode to Pure black and Day/Night mode enabled, you can still get an automatic light/pure black.

I just want to avoid adding too many options for edge-cases. I know it's just one, but the more we add, the more users looking for the most common options can't find it. I regularly get user reports asking where they can set units…

**Comment by codenyte** [comment]
> If system preference is dark, but Breezy Weather applies pure black, technically, it doesn't follow system preference

In my opinion, that's a flaw of Android. But Google doesn't want to listen to their users and they won't fix it.

> Note that if you set Dark mode to Pure black and Day/Night mode enabled, you can still get an automatic light/pure black.

Ok thanks for clarifying that, because that is my exact use case.

> I get what you're trying to achieve, but you all seemed to prefer OLED to reduce eye strain, so I find it curious that you now want both light and pure black at the same time?

I sometimes like to use scheduled light/dark mode, where dark mode only turns on at night. That's why I generally dislike it, when apps require the user to choose between 'Follow system' and pure black.

**Comment by papjul** [comment]
I looked into it a bit, and it is more complex than I thought. I sometimes can't get some surfaces to change their color, and there seems to be various places where colors are defined. There are also many types of surfaces, they must all be thoroughly tested.

But even then, it's not just replacing the background. Everything else has to take into account this new background. For example, after putting a pure black background, cards don’t look good anymore as the elevation is too important compared to the background. So it's a bit like designing a new theme, not just changing the background to pure black.

I'm tagging this as a high effort. I really can't help on this, but we will certainly need some refactoring before we can proceed to the actual implementation of the pure black. If you want to implement it, let me know what your analysis is, and what you plan to do first. Thanks!

**Comment by Username1-a** [comment]
Just in case, here's the last version 4.6.14 where you could achieve Black dark theme in Breezy Weather through Repainter app and it's comparison with the recent version 5.2.4:
![Screenshot_20240616-194949_Lawnchair](https://github.com/breezy-weather/breezy-weather/assets/47947180/d7077377-9c81-48ce-b638-9d34392764d5)
![Screenshot_20240616-195025_Lawnchair](https://github.com/breezy-weather/breezy-weather/assets/47947180/028b1169-86a7-4058-abfd-b0650cbf8e4c)
